pkg/endpoint/endpoint.go

```diff
@@ -1001,6 +1155,9 @@ func ParseEndpoint(strEp string) (*Endpoint, error) {
 
 	ep.state = StateRestoring
 
+	ep.Mutex.Lock()
```

Comment:
What is this doing?

Reply:
you know... locking things...
I think I had these when I started the sync controller in these functions (the start part needs the lock). I moved those and, clearly, didn't review my own code :/